File: bindings/python/sop/cassandra.py
import json
from typing import Dict
from . import call_go

class Cassandra:
    """
    Cassandra class provides static methods to initialize and close the global Cassandra connection.
    """

    @staticmethod
    def initialize(config: Dict[str, any]):
        """
        Initialize the global Cassandra connection.
        
        Args:
            config (Dict[str, any]): Configuration dictionary containing:
                - cluster_hosts (List[str]): List of Cassandra host addresses
                - consistency (int): Consistency level (default: LocalQuorum)
                - connection_timeout (int): Timeout in milliseconds
                - replication_clause (str): Replication strategy string
                - authenticator (Dict): Optional auth config with 'username' and 'password'
        """
        payload = json.dumps(config)
        res = call_go.open_cassandra_connection(payload)
        if res:
            raise Exception(res)

    @staticmethod
    def close():
        """Close the global Cassandra connection."""
        res = call_go.close_cassandra_connection()
        if res:
            raise Exception(res)

File: bindings/csharp/Sop/Cassandra.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Sop;

public class CassandraAuthenticator
{
    [JsonPropertyName("username")]
    public string Username { get; set; }

    [JsonPropertyName("password")]
    public string Password { get; set; }
}

public class CassandraConfig
{
    [JsonPropertyName("cluster_hosts")]
    public List<string> ClusterHosts { get; set; }

    [JsonPropertyName("consistency")]
    public int Consistency { get; set; }

    [JsonPropertyName("connection_timeout")]
    public int ConnectionTimeout { get; set; }

    [JsonPropertyName("replication_clause")]
    public string ReplicationClause { get; set; }

    [JsonPropertyName("authenticator")]
    public CassandraAuthenticator Authenticator { get; set; }
}

public static class Cassandra
{
    /// <summary>
    /// Initializes the global shared Cassandra connection.
    /// </summary>
    public static void Initialize(CassandraConfig config)
    {
        var json = System.Text.Json.JsonSerializer.Serialize(config);
        var resPtr = NativeMethods.OpenCassandraConnection(Interop.ToBytes(json));
        var res = Interop.FromPtr(resPtr);
        if (res != null) throw new SopException(res);
    }

    /// <summary>
    /// Closes the global shared Cassandra connection.
    /// </summary>
    public static void Close()
    {
        var resPtr = NativeMethods.CloseCassandraConnection();
        var res = Interop.FromPtr(resPtr);
        if (res != null) throw new SopException(res);
    }
}
